### ISSUE TITLE
This role overrides php_packages, which is also used in geerlingguy.php

### ISSUE THREAD

**Issue by mloveday** [geerlingguy/ansible-role-php-versions#46]
Our playbook has the following config defined:
```
vars:
  php_packages:
    - php{{ php_version }}-intl
    - php{{ php_version }}-zip
    ...
roles:
  - geerlingguy.php-versions
  - geerlingguy.php
  ...
```
When we run the playbook, the php packages we were specifying were not being installed. We found that `geerlingguy.php_versions` was setting `php_packages` to that role's default value, and `geerlingguy.php` was then using this default over our user-specified value. We tried specifying `php_packages` in vars as above, in a vars file, and inline with the role itself (`- { role: geerlingguy.php, vars: php_packages: ... }`), but nothing changed.

The only way we could work around this was by defining the `php_packages` variable in a new role after php-versions had run:
```
vars:
  php_packages:
    - php{{ php_version }}-cli
    - php{{ php_version }}-zip
    ...
roles:
  - geerlingguy.php-versions
  - php-packages
  - geerlingguy.php
  ...
```
... with the php-packages role containing the following ...
```
set_fact: php_packages="{{ custom_php_packages }}"
```

Firstly, is there something I'm doing wrong in specifying the `php_packages` variable? If it's something I've done wrong, what should I change and why is this particular use case different to others?

Is it just a case of renaming this role's (internal only?) variable to something more unique? If so, I'm happy to provide a PR.

Thanks!

**Comment by webarchitect609** [comment]
Today I have found this issue as affecting my playbook. I also have examined and tested the proposed solution - [PR #39](https://github.com/geerlingguy/ansible-role-php-versions/pull/39). It works with both - playbook vars and host vars.

@mloveday , as a temporary solution you can create [host vars](https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html#organizing-host-and-group-variables) to avoid overwriting variables and also avoid endless waiting for [PR #39](https://github.com/geerlingguy/ansible-role-php-versions/pull/39) to be merged. I.e. create file `~/host_vars/hostname.yml` where `~` = folder where your playbook or inventory file is; `hostname` = hostname, against which you use this role in your playbook. Reason: [PR #4](https://github.com/geerlingguy/ansible-role-php-versions/pull/4/commits/da2fa259012e77145bba58e69022503b76512d6a) let the `tasks/main.yml` file to watch for host vars only.

**Comment by webarchitect609** [comment]
@mloveday, another option is to use my fork [webarchitect609.php_versions](https://galaxy.ansible.com/webarchitect609/php_versions), where it has been fixed.

**Comment by ymhuang0808** [comment]
> @mloveday, another option is to use my fork [webarchitect609.php_versions](https://galaxy.ansible.com/webarchitect609/php_versions), where it has been fixed.

It works for me.

**Comment by pierregaste** [comment]
Same for me. Thanks for this workaround!

**Comment by webarchitect609** [comment]
@geerlingguy, I would like you to check this issue again. I believe this is a bug, formulated as: **impossible to set `php_packages` from a playbook when using `geerlingguy.php` with `geerlingguy.php-versions`**.

The fix is simple and I re-make PR #39 as a fresh PR #55 . Thank you! :)

**Comment by geerlingguy** [comment]
Fixed in #55.
